README.md
## NavigationBar React Native

Headerbar, NavigationBar support for React Native

<p align="center">
    <img src ="https://media.giphy.com/media/NlqsYiyrmGmjroqbsY/giphy.gif" />
</p>

### Content
- [NavigationBar React Native](#navigationbar-react-native)
  - [Content](#content)
  - [Installation](#installation)
  - [Usage example](#usage-example)
  - [Properties](#properties)
  - [Events](#events)
  - [License](#license)
  - [Questions?](#questions)

### Installation
```bash
npm install navigationbar-react-native
```
### Usage example
```javascript

import React, { Component } from 'react';
import {
  AppRegistry,
  StyleSheet,
  Text,Image,
  View, 
  TouchableOpacity,
} from 'react-native';
import { NavigationBar } from 'navigationbar-react-native';

const ComponentLeft = () => {
  return(
    <View style={{ flex: 1, alignItems: 'flex-start'}} >
       <TouchableOpacity style={ {justifyContent:'center', flexDirection: 'row'}}>
        <Image 
          source={require('./img/ic_back.png')}
          style={{ resizeMode: 'contain', width: 20, height: 20, alignSelf: 'center' }}
        />
        <Text style={{ color: 'white', }}>Back Home</Text>
      </TouchableOpacity>
    </View>
  );
};

const ComponentCenter = () => {
  return(
    <View style={{ flex: 1, }}>
       <Image
        source={require('./img/ic_logo.png')}
        style={{resizeMode: 'contain', width: 200, height: 35, alignSelf: 'center' }}
      />
    </View>
  );
};

const ComponentRight = () => {
  return(
    <View style={{ flex: 1, alignItems: 'flex-end', }}>
      <TouchableOpacity>
        <Text style={{ color: 'white', }}> Right </Text>
      </TouchableOpacity>
    </View>
  );
};

class App extends Component {
  render() {
    return (
      <View style={styles.container}>
        <NavigationBar 
          componentLeft     = { () =>  <ComponentLeft />   }
          componentCenter   = { () =>  <ComponentCenter /> }
          componentRight    = { () =>  <ComponentRight />  }
          navigationBarStyle= {{ backgroundColor: '#215e79' }}
          statusBarStyle    = {{ barStyle: 'light-content', backgroundColor: '#215e79' }}
        />
      </View>
    );
  }
}
```

### Properties
* `statusBarStyle` (Object) - Style is object,
* `navigationBarStyle` (Object) - Style is object,
* `componentLeft` (function) - Initial state (default:  () => <View/>),
* `componentCenter` (funtion) - Initial state (default: () => <View/>),
* `componentRight` (funtion) - Initial state (default:  () => <View/>),

### Events


### License
MIT License

### Questions?
Feel free to [create an issue](https://github.com/jundat95/navigationbar-react-native/issues)


CHANGELOG.md
# Changelog

NavigationBar React Native, Header React Native


## [v0.0.3](https://github.com/jundat95/navigationbar-react-native/releases/tag/v0.0.3) (2017/08/26)

**Implemented enhancements:**

- Add SatusBarStyle color

**Fixed bug:**


**Merged pull requests:**


## [v0.0.4](https://github.com/jundat95/navigationbar-react-native/releases/tag/v0.0.4) (2018/08/29)

**Implemented enhancements:**

- Update version support react native typescript
- Update performance code

**Fixed bug:**


**Merged pull requests:**

lib/navigation-bar.component.js
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  View, Platform,
  StatusBar,
} from 'react-native';

const MyStatusBar = ({ backgroundColor, barStyle }) => (
  <View style={[styles.statusBar, { backgroundColor }]}>
    <StatusBar
      barStyle= {barStyle}
      backgroundColor={backgroundColor}
    />
  </View>
);

export class NavigationBar extends Component {
  render() {
    let { navigationBarStyle, componentLeft, componentCenter, componentRight } = this.props;
    return (
      <View style={styles.container}>
        <MyStatusBar
          backgroundColor={this.props.statusBarStyle.backgroundColor}
          barStyle={this.props.statusBarStyle.barStyle} />
        <View style={[
          styles.naviagtionBar,
          navigationBarStyle,
        ]}>
          {componentLeft}
          {componentCenter}
          {componentRight}
        </View>
      </View>
    );
  }
}

NavigationBar.propTypes = {
  statusBarStyle: PropTypes.object,
  navigationBarStyle: PropTypes.object,
  componentLeft: PropTypes.element,
  componentCenter: PropTypes.element,
  componentRight: PropTypes.element,
};

NavigationBar.defaultProps = {
  statusBarStyle: {
    barStyle: 'light-content',
    backgroundColor: '#215e79',
  },
  navigationBarStyle: {
    backgroundColor: '#215e79',
  },
  componentLeft: <View style={{ flex: 1 }} />,
  componentCenter: <View style={{ flex: 1 }} />,
  componentRight: <View style={{ flex: 1 }} />,
};


const StatusBar_Height = Platform.OS === 'ios' ? 20 : 0;
const AppBar_Height = Platform.OS === 'ios' ? 44 : 56;

const styles = StyleSheet.create({
  container: {
  },
  statusBar: {
    height: StatusBar_Height,
  },
  naviagtionBar: {
    flexDirection: 'row',
    height: AppBar_Height,
    backgroundColor: '#215e79',
    justifyContent: 'space-between',
    alignItems: 'center',
  },
});


components/navigationbar.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const React = require("react");
const react_native_1 = require("react-native");
const MyStatusBar = (props) => (<react_native_1.View style={[styles.statusBar, props.backgroundColor]}>
    <react_native_1.StatusBar barStyle={props.barStyle} backgroundColor={props.backgroundColor}/>
  </react_native_1.View>);
class NavigationBar extends React.PureComponent {
    constructor(props) {
        super(props);
    }
    render() {
        return (<react_native_1.View style={styles.container}>
        <MyStatusBar backgroundColor={this.props.statusBarStyle.backgroundColor} barStyle={this.props.statusBarStyle.barStyle}/>
        <react_native_1.View style={[
            styles.naviagtionBar,
            this.props.navigationBarStyle,
        ]}>
          {this.props.componentLeft()}
          {this.props.componentCenter()}
          {this.props.componentRight()}
        </react_native_1.View>
      </react_native_1.View>);
    }
}
NavigationBar.defaultProps = {
    statusBarStyle: {
        barStyle: 'light-content',
        backgroundColor: '#215e79',
    },
    navigationBarStyle: {
        backgroundColor: '#215e79',
    },
    componentLeft: () => <react_native_1.View style={{ flex: 1 }}/>,
    componentCenter: () => <react_native_1.View style={{ flex: 1 }}/>,
    componentRight: () => <react_native_1.View style={{ flex: 1 }}/>,
};
exports.default = NavigationBar;
const StatusBar_Height = react_native_1.Platform.OS === 'ios' ? 20 : 0;
const AppBar_Height = react_native_1.Platform.OS === 'ios' ? 44 : 56;
const styles = react_native_1.StyleSheet.create({
    container: {},
    statusBar: {
        height: StatusBar_Height,
    },
    naviagtionBar: {
        flexDirection: 'row',
        height: AppBar_Height,
        backgroundColor: '#215e79',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
});
